server/nodejs/resources/js/survey-edit.js

```diff
@@ -0,0 +1,381 @@
+function renderEssayQuestion(questionContainer) {
+    const optionsContainer = questionContainer.querySelector('.options');
+    const buttonContainer = questionContainer.querySelector('.button-container');
+
+    optionsContainer.innerHTML = `
+        <textarea placeholder="Enter an answer here" rows="4" cols="50" disabled class="textarea"></textarea>
+    `;
+    buttonContainer.style.display = 'none';
+}
+
+// Render rating question
+function renderRatingQuestion(questionContainer, scale = 5) {
+    const optionsContainer = questionContainer.querySelector('.options');
+    const buttonContainer = questionContainer.querySelector('.button-container');
+
+    optionsContainer.innerHTML = `
+        <select class="max-rating-select" onchange="updateRatingScale(this)">
+            ${[...Array(10)].map((_, i) =>
+        `<option value="${i+1}" ${i+1 === scale ? 'selected' : ''}>${i+1}</option>`
+    ).join('')}
+        </select>
+        <div class="rating-container">
+            ${[...Array(scale)].map((_, i) =>
+        `<label>
+                    <input type="radio" name="rating" value="${i+1}">
+                    ${i+1}
+                </label>`
+    ).join('')}
+        </div>
+    `;
+    buttonContainer.style.display = 'none';
+}
+
+// Change question type
+function changeQuestionType(selectElement) {
+    const questionContainer = selectElement.closest('.question-container');
+    const optionsContainer = questionContainer.querySelector('.options');
+    const buttonContainer = questionContainer.querySelector('.button-container');
+    const newType = selectElement.value;
+
+    // Reset options based on new type
+    switch(newType) {
+        case 'multiple_choice':
+            buttonContainer.style.display = 'block';
+            renderMultipleChoiceOptions(questionContainer);
+            break;
+        case 'essay':
+            renderEssayQuestion(questionContainer);
+            break;
+        case 'rating':
+            renderRatingQuestion(questionContainer);
+            break;
+    }
+}
+
+// Make the title editable
+function makeEditable(id) {
+    const titleElement = document.getElementById(id);
+
+    // If it's already an input, save the value and revert back
+    if (titleElement.tagName === 'INPUT') {
+        const newTitle = titleElement.value;
+        titleElement.replaceWith(createTitleElement(newTitle));  // Revert back to the title
+    } else {
+        // Create an input field for editing
+        const inputElement = document.createElement('input');
+        inputElement.type = 'text';
+        inputElement.value = titleElement.textContent;
+        inputElement.addEventListener('blur', () => {
+            const newTitle = inputElement.value;
+            titleElement.replaceWith(createTitleElement(newTitle));  // Revert back to the title
+        });
+        titleElement.replaceWith(inputElement);
+        inputElement.focus();  // Focus on the input field for immediate editing
+    }
+}
```

Comment:
Use a guard clause here, this would eliminate the indentation depth, you could do it this way

```
if (titleElement.tagName === 'INPUT'){
      const newTitle = titleElement.value;
      titleElement.replaceWith(createTitleElement(newTitle));
      return;
}

// the code inside the else block here
```